src/core/Database.cpp

```diff
@@ -519,7 +525,9 @@ bool Database::setKey(const QSharedPointer<const CompositeKey>& key, bool update
 
     QByteArray oldTransformedMasterKey = m_data.transformedMasterKey;
     QByteArray transformedMasterKey;
-    if (!key->transform(*m_data.kdf, transformedMasterKey)) {
+    if (m_blockKeyTransformation) {
```

Comment:
Why not make this an optional parameter to ```setKey```? I am generally not in favor of before/after function calls because it can lead to bugs where the "after" is never called due to programming error or early-out returns.

Reply:
I don't think the CompositeKey class should have any knowledge of this. Having a function transform() that does nothing if a flag is set is counter-intuitive. This is a higher-level feature of the Database class. I agree that it would be safer with some kind of try/finally, but that's why I added the extra assertions.

Reply:
Update: actually, you are right. The way it's implemented now, only setKey() is called in a context where we do not need transformation. My first implementation look different. The only thing I don't like about a parameter is that setKey() also has two similar parameters and the signature already looks quite messy.

Reply:
Yes I can agree it is messy to have three bools in a row, but the parameters have distinguishing names and purposes.

Reply:
I wish C++ had keyword arguments like in Python.